pkg/config/config.go

```diff
@@ -76,57 +105,128 @@ func SaveToFile(config *Config, filename string) error {
 	return nil
 }
 
-func (c *Config) Validate() error {
+func (c *Config) Validate() []error {
+	var allTheErrors []error
+	re := regexp.MustCompile("^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$")
```

Comment:
`re` is not the best vriable name for what I *think* is an email syntax regex.

Validating email addresses is like the holy war. On one hand you have people who say

> Don't do a syntax validation at all. Just look for an @ sign and try to open an SMTP connection to the target host.

and then there are people who say

> basic validation (max length, must contain @, ...) is good, but don't go overboard.

and others that say

> I once found this gigantic 3-page regex that supposedly covers all regex, and now I have printed and framed it above my bed and use it everywhere. It says "works for all e-mail addresses" on the tin, so I trust it!

Over the years I started at the bottom and worked my way up. Nowadays I would only check for an @ character and maybe the max length. Because TBH, I would have to spend an hour just reading the regex if *anything* breaks. Meaning we would technically need unit tests for this monster.

You can also look for a ready-made Go module that already has an email validation.

TL;DR: I would not have such a complicated regex in my codebase without tests, or even at all.

Reply:
All of this is *unless* the backend system (GSuite in this case) does not explicitly define a format for email addreses. Looking further down, it seems like GSuite is defining a format for telephone numbers, so it's fine and encouraged to validate the format.

Reply:
Very good points, thank you!
I have changed the address validation to check for the presence of @ and the length: https://github.com/kubermatic-labs/gman/pull/11/commits/4b47f836cb50d908839f5ffef246927836676bce#diff-091ace02b49a46f55f2586cfec89a160R260 
I hope it is better now.